server/lib/settings/migrator.ts

```diff
@@ -10,19 +10,42 @@ export const runMigrations = async (
   settings: AllSettings,
   SETTINGS_PATH: string
 ): Promise<AllSettings> => {
-  const migrations = fs
-    .readdirSync(migrationsDir)
-    .filter((file) => file.endsWith('.js') || file.endsWith('.ts'))
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    .map((file) => require(path.join(migrationsDir, file)).default);
-
   let migrated = settings;
 
   try {
+    // we read old backup and create a backup of currents settings
+    const BACKUP_PATH = SETTINGS_PATH.replace('.json', '.old.json');
+    const oldBackup = fs.existsSync(BACKUP_PATH)
+      ? fs.readFileSync(BACKUP_PATH)
+      : null;
+    fs.writeFileSync(BACKUP_PATH, JSON.stringify(settings, undefined, ' '));
+
+    const migrations = fs
+      .readdirSync(migrationsDir)
+      .filter((file) => file.endsWith('.js') || file.endsWith('.ts'));
+
```

Comment:
Same thing here as above. `fs.promises`

```suggestion
const migrations = await fs.promises.readdir(migrationsDir);
const validMigrations = migrations.filter(
      (file) => file.endsWith('.js') || file.endsWith('.ts')
    );
```